flocker/ca/_ca.py

```diff
@@ -0,0 +1,53 @@
+# Copyright Hybrid Logic Ltd.  See LICENSE file for details.
+
+"""
+Low-level logic for a certificate authority.
+
+Uses ECDSA + SHA256? Or if that turns out to be too difficult RSA 4096-bit
++ SHA 256.
+"""
+
+from twisted.internet.ssl import KeyPair
+
+
+def generate_keypair():
+    """
+    Create a new key pair.
+    """
+    #return KeyPair.generate(crypto.TYPE_ECDSA, size=something?)
+    pass
+
+
+class CertificateAuthority(object):
+    """
+    A certificate authority whose configuration is stored in a specified
+    directory.
+    """
+    def __init__(self, path):
+        """
+        :param FilePath path: Directory where private key and certificate are
+            stored.
+        """
+        # try to load private key and certificate, raise exception if that
+        # fails.
+
+    @classmethod
+    def initialize(cls, path, name):
+        """
+        Generate new private/public key pair and self-sign, then store in
+        given directory.
+
+        :param FilePath path: Directory where private key and certificate are
+            stored.
+        :param bytes name: The name of the cluster.
```

Comment:
(All of this stuff matters a ton for a high-quality CA management tool.  We might not be building that so it might not actually matter at all.  But code often tends to live longer than one plans...)